sdk/program/src/entrypoint.rs

```diff
@@ -111,12 +112,18 @@ unsafe impl std::alloc::GlobalAlloc for BumpAllocator {
         let mut pos = *pos_ptr;
         if pos == 0 {
             // First time, set starting position
-            pos = self.start + self.len;
+            pos = if let Some(pos) = self.start.checked_add(self.len) {
```

Comment:
Not sure we need these changes in the bump allocator.  If there is an overflow then alloc will hit an address translation error immediately and bail.  That might be better than failing the alloc because programs in rust are not setup well for alloc failures.

Reply:
Thanks for the feedback!  Would it make more sense to revert this commit and instead apply the module-level allow to the entire file with a comment stating the only-call-through-runtime requirement and noting that the VM is better off handling any arithmetic errors stemming from this module?